WordPress/WordPressTest/ReaderSiteServiceTests.swift

```diff
@@ -0,0 +1,92 @@
+
+        stub(condition: isHost("public-api.wordpress.com")) { request in
+            NSLog("[Warning] Received an unexpected request sent to \(String(describing: request.url))")
+            return HTTPStubsResponse(error: URLError(.notConnectedToInternet))
+        }
+        HTTPStubs.removeAllStubs()
+    }
+
+    func testFollowSiteByURL() {
+        stub(condition: isHost("test.blog")) { _ in
+            HTTPStubsResponse(data: Data(), statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/test.blog")) { _ in
+            HTTPStubsResponse(jsonObject: ["ID": 42], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/mine")) { _ in
+            HTTPStubsResponse(jsonObject: ["is_following": false], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/new")) { _ in
+            HTTPStubsResponse(jsonObject: [:], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.2/read/sites/42")) { _ in
+            HTTPStubsResponse(jsonObject: [
+                "feed_ID": 100,
+                "feed_URL": "https://test.blog/feed",
+                "post_count": 0,
+            ], statusCode: 200, headers: nil)
+        }
+        waitUntil { done in
+            self.service.followSite(by: URL(string: "https://test.blog")!) {
+                XCTAssertTrue(true)
+                done()
+            } failure: { _ in
+                XCTFail("")
+                done()
+            }
+        }
+    }
+
+    func testFollowSiteByID() {
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/mine")) { _ in
+            HTTPStubsResponse(jsonObject: ["is_following": false], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/new")) { _ in
+            HTTPStubsResponse(jsonObject: [:], statusCode: 200, headers: nil)
+        }
+        stub(condition: isPath("/rest/v1.2/read/sites/42")) { _ in
+            HTTPStubsResponse(jsonObject: [
+                "feed_ID": 100,
+                "feed_URL": "https://test.blog/feed",
+                "post_count": 0,
+            ], statusCode: 200, headers: nil)
+        }
+        waitUntil { done in
+            self.service.followSite(withID: 42) {
+                XCTAssertTrue(true)
+                done()
+            } failure: { _ in
+                XCTFail("")
+                done()
+            }
+        }
+    }
+
+    func testUnfollowSiteByID() {
+        stub(condition: isPath("/rest/v1.1/sites/42/follows/mine/delete")) { _ in
+            HTTPStubsResponse(jsonObject: [:], statusCode: 200, headers: nil)
+        }
+        waitUntil { done in
+            self.service.unfollowSite(withID: 42) {
+                XCTAssertTrue(true)
+                done()
+            } failure: { _ in
+                XCTFail("")
+                done()
+            }
+        }
```

Comment:
See above for details:

```suggestion

        waitUntil { done in
            self.service.unfollowSite(withID: 42) {
                done()
            } failure: { _ in
                XCTFail()
            }
        }
```